test/desdemona/cli_test.clj

```diff
@@ -88,3 +117,54 @@
         (is (thrown-with-msg?
              Exception (re-pattern @#'aeron/aeron-launch-error-message)
              (with-fake-launcher-side-effects (aeron/-main))))))))
+
+(deftest peers-main-tests
+  (testing "first argument must be an integer"
+    (is (thrown-with-msg?
+         NumberFormatException #"BOGUS"
+         (peers/-main "BOGUS"))))
+  (testing "happy case"
+    (let [n-peers 6
+          group (gensym)
+          peers (for [_ (range n-peers)] (gensym))
+          events (atom [])
+          fn-results {'onyx.api/start-peer-group group
+                      'onyx.api/start-peers peers}
+          redef-pairs (for [sym ['onyx.api/start-peer-group
+                                 'onyx.api/start-env
+                                 'onyx.api/start-peers
+                                 'desdemona.launcher.utils/add-shutdown-hook!
+                                 'onyx.api/shutdown-peer
+                                 'onyx.api/shutdown-peers
+                                 'onyx.api/shutdown-peer-group
+                                 'clojure.core/shutdown-agents]]
+                        [(resolve sym)
+                         (fn [& args]
+                           (let [event (into [sym] args)]
+                             (swap! events conj event)
+                             (fn-results sym)))])
```

Comment:
`fn-results` is just a way to stub out those side-effecty functions where I care about the result. Instead of actually letting them run, I'm making them return sentinel values. I store those sentinel values in the fn-results map -- this is just a lookup in that map.